lab_4_fill_words_by_ngrams/main.py

```diff
@@ -98,7 +136,35 @@ def run(self, seq_len: int, prompt: str) -> str:  # type: ignore
                 or if sequence has inappropriate length,
                 or if methods used return None.
         """
-
+        if not (isinstance(seq_len, int) and seq_len > 0
+                and isinstance(prompt, str) and prompt):
+            raise ValueError
+        encoded_text = self._word_processor.encode(prompt)
+        if not encoded_text:
+            raise ValueError
+        for _ in range(seq_len):
+            next_tokens = self._model.generate_next_token(encoded_text)
+            if next_tokens is None:
+                raise ValueError
+            if not next_tokens:
+                break
+            candidates = []
+            for word, value in next_tokens.items():
+                candidates.append((word, value))
+            sorted_candidates = sorted(candidates, key=lambda x: (x[1], x[0]), reverse=True)
+            probability = 0
+            possible_tokens = tuple()
```

Comment:
why not making it a list if it needs to be changeable?

Reply:
Fixed